src/fileimpl.cpp

```diff
@@ -440,7 +440,7 @@ makeFileReader(std::shared_ptr<const FileCompound> zimFile, offset_t offset, zsi
 
   const std::string& FileImpl::getMimeType(uint16_t idx) const
   {
-    if (idx > mimeTypes.size())
+    if (idx >= mimeTypes.size())
```

Comment:
Nice catch.

It would be nice to test this.
It should be easily doable if we create a `FileImpl` (instead of Archive) on any existing file and directly call `getMimeType`.

It would even be better to move the mimetype handling in a separated object, as we've done for direntReader/direntAccessor/direntLookup/...
But it should be done in a separated PR (I'm open to add the test in this new PR, if we do it soon)

Reply:
Creating a new unit-test for this small (trivial and obviously correct) fix doesn't fit well into this PR. Let's do it in a separate refactoring PR focusing on mimetype handling.